vllm/v1/attention/backends/flashinfer.py

```diff
@@ -0,0 +1,497 @@
+# SPDX-License-Identifier: Apache-2.0
+"""Attention layer with FlashAttention."""
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Type
+
+try:
```

Comment:
fwiw you can use `from __future__ import annotations` for types related.

we can assume that if flashinfer is choosen, then it will be available (afaik attention backend will be lazy import)

You can still keep the try/except block, but probably can just raise the exception immediately.